hastexo/migrations/0001_initial.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='Stack',
+            fields=[
+                ('id', models.AutoField(verbose_name='ID', serialize=False, auto_created=True, primary_key=True)),
+                ('name', models.CharField(max_length=64, db_index=True)),
+                ('student_id', models.CharField(max_length=40, db_index=True)),
+                ('course_id', models.CharField(max_length=50, db_index=True)),
+                ('provider', models.CharField(max_length=32)),
+                ('protocol', models.CharField(max_length=32)),
+                ('port', models.IntegerField(null=True)),
+                ('status', models.CharField(max_length=32, db_index=True)),
+                ('error_msg', models.CharField(max_length=256)),
+                ('ip', models.CharField(max_length=40)),
```

Comment:
The reason the field is size 40 is precisely in order to support IPv6, but you're right that a GenericIPAddressField is better because we get some validation for free.  I'll make the change.